h264_encoder_core/src/h264_encoder.cpp

```diff
@@ -146,50 +174,27 @@ class H264EncoderImpl
     AVDictionary * opts = nullptr;
     if (codec_name.empty()) {
       codec = avcodec_find_encoder_by_name(kDefaultHardwareCodec);
-      if (nullptr == codec || !is_codec_available(codec)) {
+      if (nullptr == codec || AWS_ERR_OK != set_param(codec) || AWS_ERR_OK != open_codec(codec, param_, opts)) {
```

Comment:
I think there's memory leak if `set_param()` succeeds and `open_codec()` fails.

Let's combine
```
nullptr == codec || AWS_ERR_OK != set_param(codec) || AWS_ERR_OK != open_codec(codec, param_, opts)
```
into a single robust function, since that exact series of checks is repeated a couple of times in this code.

Reply:
Done